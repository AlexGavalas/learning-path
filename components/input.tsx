import { forwardRef, InputHTMLAttributes, ForwardedRef } from 'react';

type InputProps = InputHTMLAttributes<HTMLInputElement>;

const InputComponent = (
    props: InputProps,
    ref: ForwardedRef<HTMLInputElement>,
) => {
    return (
        <input
            {...props}
            ref={ref}
            className="pl-2 w-full h-full border-[1px] dark:border-0 border-solid bg-transparent dark:bg-neutral-800 rounded dark:placeholder-gray-300 dark:placeholder-opacity-50"
        />
    );
};

export const Input = forwardRef(InputComponent);
