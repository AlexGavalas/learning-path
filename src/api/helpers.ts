import axios from 'axios';

export const fetchFileFromStorage = async (
    filename: string,
): Promise<string> => {
    const fileServerUrl = String(process.env.PUBLIC_FILE_SERVER_URL);

    const response = await axios.get(`${fileServerUrl}/${filename}`, {
        responseType: 'text',
    });

    return response.data as string;
};
