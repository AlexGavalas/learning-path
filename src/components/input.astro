---
const { label, ...rest } = Astro.props;
---

<label class="flex h-full w-full flex-col">
    <span>{label}</span>
    <input
        {...rest}
        class="h-1/2 rounded border-[1px] border-solid border-light-border bg-transparent pl-2 text-sm dark:border-0 dark:bg-neutral-800 dark:placeholder-gray-300 dark:placeholder-opacity-60"
    />
</label>
