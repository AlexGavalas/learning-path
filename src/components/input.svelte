<script lang="ts">
    import type { HTMLInputAttributes } from 'svelte/elements';

    interface $$Props extends HTMLInputAttributes {
        label: string;
    }

    let instance: HTMLInputElement;

    export let label: $$Props['label'];
    export let value = '';

    export function focus() {
        instance.focus();
    }
</script>

<label class="flex h-full w-full flex-col">
    <span>{label}</span>
    <input
        {...$$restProps}
        bind:value
        bind:this={instance}
        class="h-1/2 rounded border-[1px] border-solid border-light-border bg-transparent pl-2 text-sm dark:border-0 dark:bg-neutral-800 dark:placeholder-gray-300 dark:placeholder-opacity-60"
    />
</label>
