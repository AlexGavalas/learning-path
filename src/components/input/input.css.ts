import { style } from '@vanilla-extract/css';

import { vars } from '../../styles/theme.css';

export const container = style({
    display: 'grid',
});

export const input = style({
    backgroundColor: vars.color.input.background,
    borderRadius: '5px',
    border: `2px solid ${vars.color.border}`,
    fontSize: '0.8em',
    paddingLeft: vars.spacing.sm,
    '::placeholder': {
        color: vars.color.secondary,
        opacity: 0.6,
    },
});
