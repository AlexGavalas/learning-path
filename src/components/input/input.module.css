.container {
    display: grid;
    gap: var(--spacing-sm);
}

.input {
    background-color: oklch(
        from var(--color-background) calc(l + 0.05) c calc(h + 0.2)
    );
    border-radius: var(--spacing-xs);
    border: none;
    font-size: 0.9em;
    line-height: 1.7;
    padding-left: var(--spacing-sm);

    &::placeholder {
        color: oklch(from var(--color-typography-primary) calc(l + 0.02) c h);
        opacity: 0.6;
    }
}
