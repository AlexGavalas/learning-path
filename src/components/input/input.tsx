import { type ForwardedRef, type InputHTMLAttributes, forwardRef } from 'react';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
    label: string;
}

const InputComponent = (
    { label, ...props }: InputProps,
    ref: ForwardedRef<HTMLInputElement>,
) => {
    return (
        <label className="flex h-full w-full flex-col">
            <span>{label}</span>
            <input
                {...props}
                ref={ref}
                className="h-1/2 w-full rounded border-[1px] border-solid bg-transparent pl-2 dark:border-0 dark:bg-neutral-800 dark:placeholder-gray-300 dark:placeholder-opacity-50"
            />
        </label>
    );
};

export const Input = forwardRef(InputComponent);
