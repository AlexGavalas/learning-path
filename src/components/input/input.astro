---
import type { HTMLAttributes } from 'astro/types';
import * as cn from './input.css';

interface Props extends HTMLAttributes<'input'> {
    label?: string;
}

const { label, class: extraClass, ...rest } = Astro.props;
---

<div class:list={[cn.container, extraClass]}>
    {label && <label for={rest.id}>{label}</label>}
    <input {...rest} class={cn.input} />
</div>
