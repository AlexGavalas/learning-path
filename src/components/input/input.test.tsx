import { axe } from 'jest-axe';

import { renderWithUser } from '~test/helpers';
import { Input } from './input';

describe('<Input />', () => {
    it('renders', () => {
        const { container } = renderWithUser(<Input label="a label" />);

        expect(container).toMatchSnapshot();
    });

    it('is accessible', async () => {
        const { container } = renderWithUser(<Input label="a label" />);

        const a11yResults = await axe(container);

        expect(a11yResults).toHaveNoViolations();
    });
});
