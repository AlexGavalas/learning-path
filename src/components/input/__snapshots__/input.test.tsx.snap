// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders 1`] = `
<div>
  <label
    class="flex h-full w-full flex-col"
  >
    <span>
      a label
    </span>
    <input
      class="h-1/2 rounded border-[1px] border-solid border-light-border bg-transparent pl-2 text-sm dark:border-0  dark:bg-neutral-800 dark:placeholder-gray-300 dark:placeholder-opacity-60"
    />
  </label>
</div>
`;
